core/streamer.py

```
@@ -262,19 +374,50 @@ def on_eeg_data_done(self, *args, **kwargs):
+                    )
+                except ProxyForwardError:
+                    # retry 소진 — FAIL_CLOSED: 측정 즉시 중단함 (중복 발동 방지)
+                    if not self._fail_closed_triggered:
+                        self._fail_closed_triggered = True
+                        print(
+                            "[FAIL_CLOSED] proxy 전송 retry 소진 — 측정 중단함"
+                            f" (subject {self.subject_index})"
+                        )
+                        try:
+                            self.close_session()
+                        except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)